### ISSUE TITLE
Incorrect pty size on startup

### ISSUE THREAD

**Issue by kchibisov** [jwilm/alacritty#2817]
Linux
Wayland

If you start vim with `alacritty -e vim` it's size will be incorrect most of the time. You can fix it with resize event though. The issue is presented on both evlp2 and master branches. Can be workaround by inserting `std::thread::sleep_ms(100)` in the beginning of the handle_resize.

Broken vim looks like this.

![Screenshot_Mon 23 Sep 2019 04:51:49 PM MSK](https://user-images.githubusercontent.com/27620401/65440395-57d74400-de31-11e9-8aa4-054b4aad6af4.png)

**Comment by osimplex** [comment]
This behavior I can reproduce on Linux with X11.

I can also reproduce this with URxvt also, using the command:
`urxvt -e vim ~/.config/alacritty/alacritty.yml`
The problem is less prone to occur if I do:
`alacritty -e sleep 0.1 ; vim ~/.config/alacritty/alacritty.yml`
Or, on URxvt:
`urxvt -e sh -c "sleep 0.1 ; vim ~/.config/alacritty/alacritty.yml"`

Bigger the, more prone to fail.

This problem, I think, is on Vim. The [Qtile project](http://docs.qtile.org/en/latest/manual/faq.html#when-i-first-start-xterm-urxvt-rxvt-containing-an-instance-of-vim-i-see-text-and-layout-corruption-what-gives) have a reference for this kind of problem occuring on Xterm, URxvt and Rxvt.
Xterm have a flag `-wf` that can be used in such situation, when the program cannot handle 'inicializing' terminal state. Terminal should "wait for the window to be mapped the first time before starting the subprocess" ([man xterm](https://linux.die.net/man/1/xterm).

**Comment by osimplex** [comment]
Can I reproduce this bug with Neovim too. `alacritty -e nvim`.
A little different of Vim, Neovim is smaller than terminal window, and, as in bug report, after resize Neovim fits in terminal window and works as expected.

These programs, Vim an Neovim, expect that the terminal window will only call them after the terminal window is properly mapped.

**Comment by chrisduerr** [comment]
Yeah, this isn't an Alacritty bug, but instead an issue with applications running inside of Alacritty without handling size changes.

So a flag like XTerm's `-wf` is required.

**Comment by Chaitanyabsprip** [comment]
I am still facing this issue, any resolve?

**Comment by hisham-maged10** [comment]
hey I've found a fix for the issue, from my understanding it's an issue related to vim/neovim, it needs a certain signal to be sent from the terminal on start
adding this to your vimrc config file ~/.vimrc
`autocmd VimEnter * :silent exec "!kill -s SIGWINCH $PPID"`
fixes it
